### DISCUSSION TITLE
Port `prettier-plugin-packagejson`

### DISCUSSION THREAD

**Discussion by smorimoto** [biomejs/biome#941] [Ideas]
I'm actively using prettier-plugin-packagejson with Prettier right now, and it's very useful for consistently organizing the manifest file across all projects, which has become increasingly complex in recent years.

**Comment by smorimoto** [comment]
https://github.com/matzkoh/prettier-plugin-packagejson

**Comment by ematipico** [comment]
We decided to not format `package.json` because npm does format it with its own style and sorting. Also prettier does the same.

**Reply by smorimoto** [reply]
I'm not sure how right it is to continue to go the way Prettier went, but I respect your decision. Thank you for all you do!

**Reply by ZetabS** [reply]
> In v1, you can set lineWidth to 1. In v2 Biome will format package.json like prettier that lib you're referring to.
>
> Biome v1 doesn't do sorting. But Biome v2 will *provide* sorting
> _Originally posted by @ematipico in https://github.com/biomejs/biome/discussions/5807#discussioncomment-12979624_

As I understand, It seems that Biome now formats `package.json`.

What do you think about this discussion now?

**Reply by arthurfiorette** [reply]
Hey @ematipico, biome changed a bit in these last years, would you mind revisiting your opinion/reply here? Would be awesome to have a assist.actions.source.organizePackageJson

**Reply by ematipico** [reply]
Yeah, sure, feel free to create an issue. Mind though, we should provide the business logic of this action so that contributors have all the information they need to implement the new action

**Reply by Netail** [reply]
I am wondering, would https://biomejs.dev/assist/actions/use-sorted-keys/ produce the same result, but performs on the full json 🤔 Else it would kind of already be implemented and we could introduce an option to only sort certain keys, where users can add `dependencies`, `peerDependencies`, `optionalDependencies` & `devDependencies`

**Reply by arthurfiorette** [reply]
@Netail Sorting package json has some special rules that are not a 1:1 sorting keys operation.

**Reply by Netail** [reply]
Ahhh okay, feel free to open a PR :)

**Reply by arthurfiorette** [reply]
Hey @ematipico and @Netail, I've opened a PR implementing it :)

https://github.com/biomejs/biome/pull/8659

**Comment by mehulkar** [comment]
[We replaced eslint and prettier][1] with biome and this is one of the things that we lost. It's not a big deal, because the order of deps/devDeps keys doesn't matter, but I do miss it. In our monorepo, we have the same `@foo` prefix for all internal packages and this formatting rule helps see them all together when you're looking at a package.

[1]: https://twitter.com/mehulkar/status/1838807815462490480

**Reply by Netail** [reply]
💯 and could think of a few more rules, which will require us to use an extra linting tool, while Biome should be the all-in-one
